### PR TITLE
Added non ssl connections for private IPs

### LINKED ISSUE #9103: Allow non-SSL connections in RFC1918 reserved / local subnet space
Requested by @anadi in #5423 and it seems reasonable. I'm open to merging a PR for this if someone wants to take it on

---------

Mmm - I had not considered the local network case much. That's an interesting point. 
There is an RFC for internal only IP space that defines all the ranges 

https://datatracker.ietf.org/doc/html/rfc1918#section-3

```
     10.0.0.0        -   10.255.255.255  (10/8 prefix)
     172.16.0.0      -   172.31.255.255  (172.16/12 prefix)
     192.168.0.0     -   192.168.255.255 (192.168/16 prefix)
```

I suppose we could add an exception for non-SSL connections to those somehow? I have not investigated this, it may not be feasible but regardless of further investigation the first question is - @anadi - would allowing non-SSL connections to custom sync servers in those network ranges satisfy your use case?

_Originally posted by @mikehardy in https://github.com/ankidroid/Anki-Android/issues/5423#issuecomment-856299731_

### PR DESCRIPTION
**Pull request by prateek-singh-3212**
## Purpose / Description
Allow non-SSL connections in RFC1918 reserved / local subnet space

## Fixes
Fixes #9103

## Approach
Added IP ranges of private networks. 
mentioned on:- https://datatracker.ietf.org/doc/html/rfc1918#section-3

## How Has This Been Tested?

Not Yet 😔

## Learning (optional, can help others)
https://developer.android.com/training/articles/security-config#ConfigCustom
https://developer.android.com/guide/topics/manifest/application-element
https://stackoverflow.com/questions/51902629/how-to-allow-all-network-connection-types-http-and-https-in-android-9-pie
https://developer.android.com/training/articles/security-ssl

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [ ] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/res/xml/network_security_config.xml [hunk 1]
```diff
@@ -11,5 +11,8 @@
     <domain-config cleartextTrafficPermitted="true">
         <domain includeSubdomains="true">localhost</domain>
         <domain includeSubdomains="true">127.0.0.1</domain>
+        <domain includeSubdomains="true">10.0.0.0</domain>
```
Comment (david-allison-1) [inline AnkiDroid/src/main/res/xml/network_security_config.xml:14]: This doesn't look like it includes the IP ranges. Could you add documentation if I'm mistaken here

Comment (mikehardy) [reply AnkiDroid/src/main/res/xml/network_security_config.xml:14]: Yeah I just saw the same thing! I was checking the XML schema def on android developer documentation and it's all hostname-based.

In the end this does not appear to be technically possible. @david-allison-1  @prateek-singh-3212 if you read the documentation and come to the same conclusion we will have to close this and related issue as not possible https://developer.android.com/training/articles/security-config.html#domain

Comment (prateek-singh-3212) [reply AnkiDroid/src/main/res/xml/network_security_config.xml:14]: I am new to this networking topic. So, I have might be wrong. My thinking process goes like this:- 'cleartextTrafficPermitted = true' is used to allow non-ssl connection and if we know the specific domain then we can put this in <domain-config>  [check](https://developer.android.com/training/articles/security-config.html#domain-config) and if we don't know the sepcific domain then we can use [check](https://developer.android.com/training/articles/security-config.html#base-config). So i put that flag in domain config..

Comment (prateek-singh-3212) [reply AnkiDroid/src/main/res/xml/network_security_config.xml:14]: Our main aim is to allow http connection in the private IP's. So I thought we can use this way and followed  [this](https://stackoverflow.com/questions/51902629/how-to-allow-all-network-connection-types-http-and-https-in-android-9-pie)

Comment (david-allison-1) [reply AnkiDroid/src/main/res/xml/network_security_config.xml:14]: Agreed

We only want HTTP for specific private IP ranges, so having the attribute in the root config isn't acceptable

We can't specify IP ranges (to my understanding) using this method, so this won't be possible

Happy to close as "won't/can't fix"

Comment (prateek-singh-3212) [reply AnkiDroid/src/main/res/xml/network_security_config.xml:14]: Agreed we can't add specific ranges.

Comment (mikehardy) [reply AnkiDroid/src/main/res/xml/network_security_config.xml:14]: Okay, going to close this and the related issue. We definitely tried! Thanks @prateek-singh-3212 !

**Pull request closed without merging**